<a href="https://colab.research.google.com/github/monica-millay/final-assignment/blob/main/Evictions_in_Bushwick_data_cleaning.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Welcome!
This is the Python notebook where I will clean the data for my [Evictions in Bushwick](https://monica-millay.github.io/final-assignment/) Mapbox map!

#Step 1. Install packages needed to clean data

In [ ]:
#install pydrive
!pip install -U -q pydrive2

from pydrive2.auth import GoogleAuth
from pydrive2.drive import GoogleDrive
from google.colab import auth
from oauth2client.client import GoogleCredentials


# Authenticate and create the PyDrive client.
auth.authenticate_user()
gauth = GoogleAuth()
gauth.credentials = GoogleCredentials.get_application_default()
drive = GoogleDrive(gauth)
print("Can now run Python in Google drive!")

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.7/46.7 kB 1.5 MB/s eta 0:00:00
Can now run Python in Google drive!


In [ ]:
#install packages (pandas, Socrata for API)
import requests
import pandas as pd
!pip install pandas geopandas
import geopandas as gpd
!pip install sodapy
from sodapy import Socrata
print("Packages imported!")

Packages imported!


#Step 2. Pull in eviction data from NYC OpenData API

**Note eviction data is filtered to just Neighborhood Tabulation Areas (NTAs) of Bushwick North and Bushwick South**

In [ ]:
#BUSHWICK NORTH

#'None' in place of application token, and no username or password bc public dataset
client = Socrata("data.cityofnewyork.us", None)

#filter to just "Bushwick North"
results = client.get("6z8x-wfk4", nta="Bushwick North")

#convert to pandas DataFrame
evictions_bushwick_north = pd.DataFrame.from_records(results)

In [ ]:
#BUSHWICK SOUTH

#'None' in place of application token, and no username or password bc public dataset
client = Socrata("data.cityofnewyork.us", None)

#filter to just "Bushwick South"
results = client.get("6z8x-wfk4", nta="Bushwick South")

#convert to pandas DataFrame
evictions_bushwick_south = pd.DataFrame.from_records(results)

In [ ]:
evictions_bushwick_south.head()

court_index_number docket_number     eviction_address eviction_apt_num  \
0         K303483/22        105174   1184 HALSEY STREET            APT 1   
1          K58364/18        091161  180 MESEROLE STREET               2L   
2          305267/21        022001  186 SCHAEFER STREET               2R   
3          312450/22        109003  248 MCKIBBIN STREET               1T   
4         0316061/22        110285  618 BUSHWICK AVENUE              116   

             executed_date marshal_first_name marshal_last_name  \
0  2022-10-17T00:00:00.000             Ileana            Rivera   
1  2018-09-27T00:00:00.000             Ileana            Rivera   
2  2022-12-09T00:00:00.000             George       Essock, Jr.   
3  2022-08-11T00:00:00.000             Justin          Grossman   
4  2023-05-17T00:00:00.000              Henry             Daley   

  residential_commercial_ind   borough eviction_zip         ejectment  \
0                Residential  BROOKLYN        11207  Not an Ejectment   
1                Residential  BROOKLYN        11206  Not an Ejectment   
2                Residential  BROOKLYN        11207  Not an Ejectment   
3                Residential  BROOKLYN        11206  Not an Ejectment   
4                Residential  BROOKLYN        11206  Not an Ejectment   

  eviction_possession   latitude   longitude community_board council_district  \
0          Possession  40.690934  -73.910096               4               37   
1          Possession  40.708170  -73.942185               1               34   
2          Possession  40.689620  -73.907866               4               37   
3          Possession  40.705604  -73.937578               1               34   
4          Possession  40.698207  -73.933899               4               34   

  census_tract      bin         bbl             nta  
0          413  3078836  3034110027  Bushwick South  
1          493  3071141  3030530016  Bushwick South  
2          411  3079494  3034290024  Bushwick South  
3          485  3071420  3030910027  Bushwick South  
4          389  3425505  3031700026  Bushwick South

In [ ]:
#combine Bushwick North and Bushwick South dataframes into one Bushwick dataframe
all_bushwick_data = [evictions_bushwick_north,evictions_bushwick_south]
evictions_bushwick = pd.concat(all_bushwick_data)
evictions_bushwick

court_index_number docket_number      eviction_address eviction_apt_num  \
0             90608/16        020761   165 STANHOPE STREET               1L   
1             75311/18        084322  476 JEFFERSON STREET              420   
2            324543/22        173822  832 KNICKERBOCKER AV               3F   
3            K64838/17        083108     49 WYCKOFF AVENUE         UNIT 2-B   
4            330207/22        031342     1461 GATES AVENUE              1-B   
..                 ...           ...                   ...              ...   
775           55733/18        077990    991 WILLOUGHBY AVE               8C   
776         310933/22K        109803     182 WILSON AVENUE           3-LEFT   
777          318567/22        113343  1330 BUSHWICK AVENUE                3   
778           95388/17         17593    50 STANHOPE STREET               1R   
779           61670/18        012113     608 WILSON AVENUE                1   

               executed_date marshal_first_name marshal_last_name  \
0    2017-01-20T00:00:00.000               Gary              Rose   
1    2018-09-28T00:00:00.000             Justin          Grossman   
2    2023-09-12T00:00:00.000             Alfred          Locascio   
3    2017-11-29T00:00:00.000             Ileana            Rivera   
4    2023-08-07T00:00:00.000               Gary              Rose   
..                       ...                ...               ...   
775  2018-04-30T00:00:00.000             Justin          Grossman   
776  2023-12-06T00:00:00.000             Justin          Grossman   
777  2023-02-27T00:00:00.000             Justin          Grossman   
778  2018-08-29T00:00:00.000             Edward             Guida   
779  2018-10-09T00:00:00.000             George       Essock, Jr.   

    residential_commercial_ind   borough eviction_zip         ejectment  \
0                  Residential  BROOKLYN        11237  Not an Ejectment   
1                  Residential  BROOKLYN        11237  Not an Ejectment   
2                  Residential  BROOKLYN        11207  Not an Ejectment   
3                  Residential  BROOKLYN        11237  Not an Ejectment   
4                  Residential  BROOKLYN        11237  Not an Ejectment   
..                         ...       ...          ...               ...   
775                Residential  BROOKLYN        11221  Not an Ejectment   
776                Residential  BROOKLYN        11237  Not an Ejectment   
777                Residential  BROOKLYN        11207  Not an Ejectment   
778                Residential  BROOKLYN        11221  Not an Ejectment   
779                Residential  BROOKLYN        11207  Not an Ejectment   

    eviction_possession   latitude   longitude community_board  \
0            Possession  40.699213  -73.921910               4   
1            Possession  40.708696  -73.921870               4   
2            Possession  40.691985  -73.906381               4   
3            Possession  40.705921  -73.921794               4   
4            Possession  40.697607  -73.914112               4   
..                  ...        ...         ...             ...   
775          Possession  40.698696  -73.929254               4   
776          Possession  40.699516  -73.923652               4   
777          Possession  40.685976  -73.911967               4   
778          Possession  40.695905  -73.925131               4   
779          Possession  40.689850  -73.906661               4   

    council_district census_tract      bin         bbl             nta  
0                 37          431  3074168  3032570051  Bushwick North  
1                 34          447  3072427  3031780020  Bushwick North  
2                 37          409  3342357  3034240035  Bushwick North  
3                 34          447  3072788  3032000001  Bushwick North  
4                 37          433  3000000  3033350050  Bushwick North  
..               ...          ...      ...         ...             ...  
775               34 

In [ ]:
#check for missing values

def missing_values_table(df):
    mis_val = df.isnull().sum()
    mis_val_percent = 100 * df.isnull().sum() / len(df)
    mis_val_table = pd.concat([mis_val, mis_val_percent], axis=1)
    mis_val_table_ren_columns = mis_val_table.rename(
    columns = {0 : 'Missing Values', 1 : '% of Total Values'})
    mis_val_table_ren_columns = mis_val_table_ren_columns[
        mis_val_table_ren_columns.iloc[:,1] != 0].sort_values(
    '% of Total Values', ascending=False).round(1)
    print ("Your selected dataframe has " + str(df.shape[1]) + " columns.\n"
        "There are " + str(mis_val_table_ren_columns.shape[0]) +
            " columns that have missing values.")
    return mis_val_table_ren_columns

missing_values_table(evictions_bushwick)

Your selected dataframe has 20 columns.
There are 3 columns that have missing values.


Missing Values  % of Total Values
eviction_apt_num              99                8.4
bin                            1                0.1
bbl                            1                0.1

#Step 3. Filter to just residential evictions from 2017-2023

**I'm dropping observations from 2024, because I want only full years' worth of data for my map**

In [ ]:
#keep only residential evictions
evictions_bushwick = evictions_bushwick[evictions_bushwick['residential_commercial_ind'] == 'Residential']

In [ ]:
#drop if the eviction is still pending (i.e., there's no date for when the eviction was executed)
#there shouldn't be any observations that meet this criteria, since I had no values for this column when I ran missing_values_table
evictions_bushwick.dropna(subset=['executed_date'])

court_index_number docket_number      eviction_address eviction_apt_num  \
0             90608/16        020761   165 STANHOPE STREET               1L   
1             75311/18        084322  476 JEFFERSON STREET              420   
2            324543/22        173822  832 KNICKERBOCKER AV               3F   
3            K64838/17        083108     49 WYCKOFF AVENUE         UNIT 2-B   
4            330207/22        031342     1461 GATES AVENUE              1-B   
..                 ...           ...                   ...              ...   
775           55733/18        077990    991 WILLOUGHBY AVE               8C   
776         310933/22K        109803     182 WILSON AVENUE           3-LEFT   
777          318567/22        113343  1330 BUSHWICK AVENUE                3   
778           95388/17         17593    50 STANHOPE STREET               1R   
779           61670/18        012113     608 WILSON AVENUE                1   

               executed_date marshal_first_name marshal_last_name  \
0    2017-01-20T00:00:00.000               Gary              Rose   
1    2018-09-28T00:00:00.000             Justin          Grossman   
2    2023-09-12T00:00:00.000             Alfred          Locascio   
3    2017-11-29T00:00:00.000             Ileana            Rivera   
4    2023-08-07T00:00:00.000               Gary              Rose   
..                       ...                ...               ...   
775  2018-04-30T00:00:00.000             Justin          Grossman   
776  2023-12-06T00:00:00.000             Justin          Grossman   
777  2023-02-27T00:00:00.000             Justin          Grossman   
778  2018-08-29T00:00:00.000             Edward             Guida   
779  2018-10-09T00:00:00.000             George       Essock, Jr.   

    residential_commercial_ind   borough eviction_zip         ejectment  \
0                  Residential  BROOKLYN        11237  Not an Ejectment   
1                  Residential  BROOKLYN        11237  Not an Ejectment   
2                  Residential  BROOKLYN        11207  Not an Ejectment   
3                  Residential  BROOKLYN        11237  Not an Ejectment   
4                  Residential  BROOKLYN        11237  Not an Ejectment   
..                         ...       ...          ...               ...   
775                Residential  BROOKLYN        11221  Not an Ejectment   
776                Residential  BROOKLYN        11237  Not an Ejectment   
777                Residential  BROOKLYN        11207  Not an Ejectment   
778                Residential  BROOKLYN        11221  Not an Ejectment   
779                Residential  BROOKLYN        11207  Not an Ejectment   

    eviction_possession   latitude   longitude community_board  \
0            Possession  40.699213  -73.921910               4   
1            Possession  40.708696  -73.921870               4   
2            Possession  40.691985  -73.906381               4   
3            Possession  40.705921  -73.921794               4   
4            Possession  40.697607  -73.914112               4   
..                  ...        ...         ...             ...   
775          Possession  40.698696  -73.929254               4   
776          Possession  40.699516  -73.923652               4   
777          Possession  40.685976  -73.911967               4   
778          Possession  40.695905  -73.925131               4   
779          Possession  40.689850  -73.906661               4   

    council_district census_tract      bin         bbl             nta  
0                 37          431  3074168  3032570051  Bushwick North  
1                 34          447  3072427  3031780020  Bushwick North  
2                 37          409  3342357  3034240035  Bushwick North  
3                 34          447  3072788  3032000001  Bushwick North  
4                 37          433  3000000  3033350050  Bushwick North  
..               ...          ...      ...         ...             ...  
775               34 

In [ ]:
#pull out year from executed_date column
from datetime import datetime

def extract_year(date_str):
    return pd.to_datetime(date_str).year

evictions_bushwick['Year'] = evictions_bushwick['executed_date'].apply(extract_year)

<ipython-input-12-39779c20b6b3>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  evictions_bushwick['Year'] = evictions_bushwick['executed_date'].apply(extract_year)


court_index_number docket_number      eviction_address eviction_apt_num  \
0           90608/16        020761   165 STANHOPE STREET               1L   
1           75311/18        084322  476 JEFFERSON STREET              420   
2          324543/22        173822  832 KNICKERBOCKER AV               3F   
3          K64838/17        083108     49 WYCKOFF AVENUE         UNIT 2-B   
4          330207/22        031342     1461 GATES AVENUE              1-B   

             executed_date marshal_first_name marshal_last_name  \
0  2017-01-20T00:00:00.000               Gary              Rose   
1  2018-09-28T00:00:00.000             Justin          Grossman   
2  2023-09-12T00:00:00.000             Alfred          Locascio   
3  2017-11-29T00:00:00.000             Ileana            Rivera   
4  2023-08-07T00:00:00.000               Gary              Rose   

  residential_commercial_ind   borough eviction_zip  ... eviction_possession  \
0                Residential  BROOKLYN        11237  ...          Possession   
1                Residential  BROOKLYN        11237  ...          Possession   
2                Residential  BROOKLYN        11207  ...          Possession   
3                Residential  BROOKLYN        11237  ...          Possession   
4                Residential  BROOKLYN        11237  ...          Possession   

    latitude   longitude community_board council_district census_tract  \
0  40.699213  -73.921910               4               37          431   
1  40.708696  -73.921870               4               34          447   
2  40.691985  -73.906381               4               37          409   
3  40.705921  -73.921794               4               34          447   
4  40.697607  -73.914112               4               37          433   

       bin         bbl             nta  Year  
0  3074168  3032570051  Bushwick North  2017  
1  3072427  3031780020  Bushwick North  2018  
2  3342357  3034240035  Bushwick North  2023  
3  3072788  3032000001  Bushwick North  2017  
4  3000000  3033350050  Bushwick North  2023  

[5 rows x 21 columns]

In [ ]:
evictions_bushwick['Year'] = evictions_bushwick['Year'].astype(int)

<ipython-input-13-06c3d54063c1>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  evictions_bushwick['Year'] = evictions_bushwick['Year'].astype(int)


In [ ]:
#now drop observations from 2024
evictions_bushwick = evictions_bushwick.drop(evictions_bushwick[evictions_bushwick.Year == 2024].index)

In [ ]:
#sort descending by date to make sure all 2024 observations were dropped
evictions_bushwick.sort_values(by=['executed_date'],ascending=False,inplace=True)
evictions_bushwick.head()

court_index_number docket_number                   eviction_address  \
188          312824/22         17317       325 KNICKERBOCKER AVE APT 2L   
374          309346/23         17887  255 MCKIBBIN STREET APARTMENT#211   
121         K312969/23        110686                1346 HANCOCK STREET   
547           53656/20        115429                 57 TROUTMAN STREET   
314          318172/23        124268                 156 CENTRAL AVENUE   

    eviction_apt_num            executed_date marshal_first_name  \
188               2L  2023-12-21T00:00:00.000             Robert   
374              211  2023-12-19T00:00:00.000             Robert   
121               1L  2023-12-18T00:00:00.000             Ileana   
547               2L  2023-12-15T00:00:00.000             Justin   
314        1ST FLOOR  2023-12-15T00:00:00.000             Justin   

    marshal_last_name residential_commercial_ind   borough eviction_zip  ...  \
188          Renzulli                Residential  BROOKLYN        11237  ...   
374          Renzulli                Residential  BROOKLYN        11206  ...   
121            Rivera                Residential  BROOKLYN        11237  ...   
547          Grossman                Residential  BROOKLYN        11206  ...   
314          Grossman                Residential  BROOKLYN        11221  ...   

    eviction_possession   latitude   longitude community_board  \
188          Possession  40.701310  -73.922727               4   
374          Possession  40.705656  -73.937397               1   
121          Possession  40.694562  -73.908876               4   
547          Possession  40.698470  -73.931825               4   
314          Possession  40.699452  -73.927576               4   

    council_district census_tract      bin         bbl             nta  Year  
188               37          429  3073584  3032360004  Bushwick North  2023  
374               34          485  3071385  3030820065  Bushwick South  2023  
121               37          437  3078388  3033990021  Bushwick North  2023  
547               34          391  3072222  3031710055  Bushwick South  2023  
314               34          423  3348124  3032070132  Bushwick South  2023  

[5 rows x 21 columns]

#Step 4. Delete duplicates

**Tenants can get a stay of eviction even after the Marshall's Notice is issued.**

If records have the same value for court_index_number (i.e., the same plaintiff is being sued in Housing Court to be evicted from the same apartment), I will keep only the most recent record. That way, an eviction that was stayed and then executed won't show up as two separate evictions.


In [ ]:
#find dupes for court_index_number and keep first (I already descending sorted by date, so the most recent records will be first.)
evictions_bushwick.drop_duplicates(subset='court_index_number', keep='first', inplace=True)

#Step 5. Clean data to use in map and export

**Create one column for address.**

I tried cleaning up the apartment number column, but it's super inconsistent. (Some values are 1ST FLOOR versus 2L verus 211, so it's hard to get the capitalization right.)

Also, sometimes apartment number is in the address line too. If I try to concetenate the address, it will turn into something like "325 Knickerbocker Ave Apt 2L, 2L, Brooklyn, NY 11237"

**I'm going to drop apartment number from the Address column, so I can focus on making my map (and not on trying to clean this column).**

In [ ]:
#change capitalization sytle to just first letter capitalized for street address
evictions_bushwick['Address'] = evictions_bushwick['eviction_address'].str.title()
evictions_bushwick.head()

court_index_number docket_number                   eviction_address  \
188          312824/22         17317       325 KNICKERBOCKER AVE APT 2L   
374          309346/23         17887  255 MCKIBBIN STREET APARTMENT#211   
121         K312969/23        110686                1346 HANCOCK STREET   
547           53656/20        115429                 57 TROUTMAN STREET   
314          318172/23        124268                 156 CENTRAL AVENUE   

    eviction_apt_num            executed_date marshal_first_name  \
188               2L  2023-12-21T00:00:00.000             Robert   
374              211  2023-12-19T00:00:00.000             Robert   
121               1L  2023-12-18T00:00:00.000             Ileana   
547               2L  2023-12-15T00:00:00.000             Justin   
314        1ST FLOOR  2023-12-15T00:00:00.000             Justin   

    marshal_last_name residential_commercial_ind   borough eviction_zip  ...  \
188          Renzulli                Residential  BROOKLYN        11237  ...   
374          Renzulli                Residential  BROOKLYN        11206  ...   
121            Rivera                Residential  BROOKLYN        11237  ...   
547          Grossman                Residential  BROOKLYN        11206  ...   
314          Grossman                Residential  BROOKLYN        11221  ...   

      latitude   longitude community_board council_district census_tract  \
188  40.701310  -73.922727               4               37          429   
374  40.705656  -73.937397               1               34          485   
121  40.694562  -73.908876               4               37          437   
547  40.698470  -73.931825               4               34          391   
314  40.699452  -73.927576               4               34          423   

         bin         bbl             nta  Year  \
188  3073584  3032360004  Bushwick North  2023   
374  3071385  3030820065  Bushwick South  2023   
121  3078388  3033990021  Bushwick North  2023   
547  3072222  3031710055  Bushwick South  2023   
314  3348124  3032070132  Bushwick South  2023   

                               Address  
188       325 Knickerbocker Ave Apt 2L  
374  255 Mckibbin Street Apartment#211  
121                1346 Hancock Street  
547                 57 Troutman Street  
314                 156 Central Avenue  

[5 rows x 22 columns]

In [ ]:
#create one column for full address -- without apartment number :(
evictions_bushwick['Full-Address'] = evictions_bushwick['Address'] + ', Brooklyn, NY ' + evictions_bushwick['eviction_zip']
evictions_bushwick.head()

court_index_number docket_number                   eviction_address  \
188          312824/22         17317       325 KNICKERBOCKER AVE APT 2L   
374          309346/23         17887  255 MCKIBBIN STREET APARTMENT#211   
121         K312969/23        110686                1346 HANCOCK STREET   
547           53656/20        115429                 57 TROUTMAN STREET   
314          318172/23        124268                 156 CENTRAL AVENUE   

    eviction_apt_num            executed_date marshal_first_name  \
188               2L  2023-12-21T00:00:00.000             Robert   
374              211  2023-12-19T00:00:00.000             Robert   
121               1L  2023-12-18T00:00:00.000             Ileana   
547               2L  2023-12-15T00:00:00.000             Justin   
314        1ST FLOOR  2023-12-15T00:00:00.000             Justin   

    marshal_last_name residential_commercial_ind   borough eviction_zip  ...  \
188          Renzulli                Residential  BROOKLYN        11237  ...   
374          Renzulli                Residential  BROOKLYN        11206  ...   
121            Rivera                Residential  BROOKLYN        11237  ...   
547          Grossman                Residential  BROOKLYN        11206  ...   
314          Grossman                Residential  BROOKLYN        11221  ...   

      longitude community_board council_district census_tract      bin  \
188  -73.922727               4               37          429  3073584   
374  -73.937397               1               34          485  3071385   
121  -73.908876               4               37          437  3078388   
547  -73.931825               4               34          391  3072222   
314  -73.927576               4               34          423  3348124   

            bbl             nta  Year                            Address  \
188  3032360004  Bushwick North  2023       325 Knickerbocker Ave Apt 2L   
374  3030820065  Bushwick South  2023  255 Mckibbin Street Apartment#211   
121  3033990021  Bushwick North  2023                1346 Hancock Street   
547  3031710055  Bushwick South  2023                 57 Troutman Street   
314  3032070132  Bushwick South  2023                 156 Central Avenue   

                                          Full-Address  
188   325 Knickerbocker Ave Apt 2L, Brooklyn, NY 11237  
374  255 Mckibbin Street Apartment#211, Brooklyn, N...  
121            1346 Hancock Street, Brooklyn, NY 11237  
547             57 Troutman Street, Brooklyn, NY 11206  
314             156 Central Avenue, Brooklyn, NY 11221  

[5 rows x 23 columns]

In [ ]:
#convert 'date' column to simple datetime type
evictions_bushwick['executed_date'] = pd.to_datetime(evictions_bushwick['executed_date'])

#create a new column with formatted dates (month written out, day, year)
evictions_bushwick['Date'] = evictions_bushwick['executed_date'].dt.strftime('%B %d, %Y')
evictions_bushwick.head()

court_index_number docket_number                   eviction_address  \
188          312824/22         17317       325 KNICKERBOCKER AVE APT 2L   
374          309346/23         17887  255 MCKIBBIN STREET APARTMENT#211   
121         K312969/23        110686                1346 HANCOCK STREET   
547           53656/20        115429                 57 TROUTMAN STREET   
314          318172/23        124268                 156 CENTRAL AVENUE   

    eviction_apt_num executed_date marshal_first_name marshal_last_name  \
188               2L    2023-12-21             Robert          Renzulli   
374              211    2023-12-19             Robert          Renzulli   
121               1L    2023-12-18             Ileana            Rivera   
547               2L    2023-12-15             Justin          Grossman   
314        1ST FLOOR    2023-12-15             Justin          Grossman   

    residential_commercial_ind   borough eviction_zip  ... community_board  \
188                Residential  BROOKLYN        11237  ...               4   
374                Residential  BROOKLYN        11206  ...               1   
121                Residential  BROOKLYN        11237  ...               4   
547                Residential  BROOKLYN        11206  ...               4   
314                Residential  BROOKLYN        11221  ...               4   

    council_district census_tract      bin         bbl             nta  Year  \
188               37          429  3073584  3032360004  Bushwick North  2023   
374               34          485  3071385  3030820065  Bushwick South  2023   
121               37          437  3078388  3033990021  Bushwick North  2023   
547               34          391  3072222  3031710055  Bushwick South  2023   
314               34          423  3348124  3032070132  Bushwick South  2023   

                               Address  \
188       325 Knickerbocker Ave Apt 2L   
374  255 Mckibbin Street Apartment#211   
121                1346 Hancock Street   
547                 57 Troutman Street   
314                 156 Central Avenue   

                                          Full-Address               Date  
188   325 Knickerbocker Ave Apt 2L, Brooklyn, NY 11237  December 21, 2023  
374  255 Mckibbin Street Apartment#211, Brooklyn, N...  December 19, 2023  
121            1346 Hancock Street, Brooklyn, NY 11237  December 18, 2023  
547             57 Troutman Street, Brooklyn, NY 11206  December 15, 2023  
314             156 Central Avenue, Brooklyn, NY 11221  December 15, 2023  

[5 rows x 24 columns]

In [ ]:
#add indicator for if an eviction occurred during the COVID-19 Moratorium
evictions_bushwick['During-Moratorium'] = (
    (evictions_bushwick['executed_date'] >= '2020-03-07') & (evictions_bushwick['executed_date'] <= '2022-01-15')).astype(int)
evictions_bushwick.head()

court_index_number docket_number                   eviction_address  \
188          312824/22         17317       325 KNICKERBOCKER AVE APT 2L   
374          309346/23         17887  255 MCKIBBIN STREET APARTMENT#211   
121         K312969/23        110686                1346 HANCOCK STREET   
547           53656/20        115429                 57 TROUTMAN STREET   
314          318172/23        124268                 156 CENTRAL AVENUE   

    eviction_apt_num executed_date marshal_first_name marshal_last_name  \
188               2L    2023-12-21             Robert          Renzulli   
374              211    2023-12-19             Robert          Renzulli   
121               1L    2023-12-18             Ileana            Rivera   
547               2L    2023-12-15             Justin          Grossman   
314        1ST FLOOR    2023-12-15             Justin          Grossman   

    residential_commercial_ind   borough eviction_zip  ... council_district  \
188                Residential  BROOKLYN        11237  ...               37   
374                Residential  BROOKLYN        11206  ...               34   
121                Residential  BROOKLYN        11237  ...               37   
547                Residential  BROOKLYN        11206  ...               34   
314                Residential  BROOKLYN        11221  ...               34   

    census_tract      bin         bbl             nta  Year  \
188          429  3073584  3032360004  Bushwick North  2023   
374          485  3071385  3030820065  Bushwick South  2023   
121          437  3078388  3033990021  Bushwick North  2023   
547          391  3072222  3031710055  Bushwick South  2023   
314          423  3348124  3032070132  Bushwick South  2023   

                               Address  \
188       325 Knickerbocker Ave Apt 2L   
374  255 Mckibbin Street Apartment#211   
121                1346 Hancock Street   
547                 57 Troutman Street   
314                 156 Central Avenue   

                                          Full-Address               Date  \
188   325 Knickerbocker Ave Apt 2L, Brooklyn, NY 11237  December 21, 2023   
374  255 Mckibbin Street Apartment#211, Brooklyn, N...  December 19, 2023   
121            1346 Hancock Street, Brooklyn, NY 11237  December 18, 2023   
547             57 Troutman Street, Brooklyn, NY 11206  December 15, 2023   
314             156 Central Avenue, Brooklyn, NY 11221  December 15, 2023   

    During-Moratorium  
188                 0  
374                 0  
121                 0  
547                 0  
314                 0  

[5 rows x 25 columns]

In [ ]:
evictions_bushwick = evictions_bushwick.rename(
    {'longitude': 'Longitude',
     'latitude':'Latitude'}, axis="columns"
)

In [ ]:
#keep fields that I will need for map -- Address, Date, Year, Longitude, Latitude
evictions_bushwick = evictions_bushwick[['Date','Address','Year','Longitude','Latitude']]
evictions_bushwick.head()

Date                            Address  \
188  December 21, 2023       325 Knickerbocker Ave Apt 2L   
374  December 19, 2023  255 Mckibbin Street Apartment#211   
121  December 18, 2023                1346 Hancock Street   
547  December 15, 2023                 57 Troutman Street   
314  December 15, 2023                 156 Central Avenue   

                                          Full-Address  Year  \
188   325 Knickerbocker Ave Apt 2L, Brooklyn, NY 11237  2023   
374  255 Mckibbin Street Apartment#211, Brooklyn, N...  2023   
121            1346 Hancock Street, Brooklyn, NY 11237  2023   
547             57 Troutman Street, Brooklyn, NY 11206  2023   
314             156 Central Avenue, Brooklyn, NY 11221  2023   

     During-Moratorium   Longitude   Latitude  
188                  0  -73.922727  40.701310  
374                  0  -73.937397  40.705656  
121                  0  -73.908876  40.694562  
547                  0  -73.931825  40.698470  
314                  0  -73.927576  40.699452

In [ ]:
#download geojson for map
gdf = gpd.GeoDataFrame(evictions_bushwick, geometry=gpd.points_from_xy(evictions_bushwick['Longitude'], evictions_bushwick['Latitude']))
gdf.to_file('evictions_bushwick_cleaned.geojson', driver='GeoJSON')
files.download('evictions_bushwick_cleaned.geojson')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
#count evictions per year for line chart
yearly_counts = evictions_bushwick['Year'].value_counts().reset_index()
yearly_counts.columns = ['Year', 'Count']
yearly_counts.sort_values(by=['Year'])

In [ ]:
#download csv file to make line chart
from google.colab import files
evictions_bushwick.to_csv('evictions_bushwick_cleaned.csv',index=False)
files.download('evictions_bushwick_cleaned.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>